app/subscriber/src/components/content-list/ContentList.tsx

```diff
@@ -0,0 +1,88 @@
+import moment from 'moment';
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Checkbox, Col, IContentModel, Row, Show } from 'tno-core';
+
+import { IGroupByState, IToggleStates } from './interfaces';
+import * as styled from './styled';
+import { determineToneIcon, groupContent, truncateTeaser } from './utils';
+
+export interface IContentListProps {
+  /** content is an array of content objects to be displayed and manipulated by the content list*/
+  content: IContentModel[];
+  /** toggleStates is an object that contains the state of the toggleable content information */
+  toggleStates: IToggleStates;
+  /** groupBy is the key by which the content will be grouped */
+  groupBy: IGroupByState;
+  /** determine the selected content based on the checkbox */
+  setSelected: React.Dispatch<React.SetStateAction<IContentModel[]>>;
+  /** array of selected content */
+  selected: IContentModel[];
+}
+
+export const ContentList: React.FC<IContentListProps> = ({
+  content,
+  toggleStates,
+  groupBy = 'source',
+  setSelected,
+  selected,
+}) => {
+  const navigate = useNavigate();
+  const grouped = groupContent(groupBy, content);
+
+  const handleCheckboxChange = (item: IContentModel, isChecked: boolean) => {
```

Comment:
Depending on dependencies it often is best to always apply `React.useCallback` for functions within a component, otherwise they will result in a rerender of the elements that use them.